src/codegeneration/ImportRuntimeTrait.js

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {CONST, VAR} from '../syntax/TokenType.js';
+import {
+  ImportDeclaration,
+  ImportedBinding,
+  ImportSpecifier,
+  ImportSpecifierSet,
+  Module,
+  ModuleSpecifier,
+  Script,
+} from '../syntax/trees/ParseTrees.js';
+import {StringSet} from '../util/StringSet.js';
+import {
+  createBindingIdentifier,
+  createIdentifierToken,
+  createIdentifierExpression,
+  createMemberExpression,
+  createStringLiteral,
+  createStringLiteralToken,
+  createVariableStatement,
+} from './ParseTreeFactory.js';
+import {parseExpression} from './PlaceholderParser.js';
+import {prependStatements} from './PrependStatements.js';
+
+function toTempName(name) {
+  return `$__${name}`;
+}
+
+function getDeclarationType(options) {
+  return options.parseOptions.blockBinding &&
+      !options.transformOptions.blockBinding ? CONST : VAR;
+}
+
+export default function ImportRuntimeTrait(ParseTreeTransformerClass) {
```

Comment:
I gather that this Trait will prepend runtime imports iff option is true (obvious) and the ParseTreeTransformerClass needs the import as indicated by `tree !== transformed`.  The latter is not obvious so I think a comment here or in the top level transform functions would help,